### PR TITLE
Fix build issues around SDK change 

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @jasonmalinowski, @dpoeschl, @CyrusNajmabadi, @Pilchie, @agocke 

Alternate approach which actually fixes the SelfContained issue completely.  If this passes will merge this one in favor of the other.

**Comment by jaredpar** [comment]
test mac please

File: src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj [hunk 1]
```diff
@@ -10,9 +10,10 @@
     <Prefer32Bit>false</Prefer32Bit>
     <LargeAddressAware>true</LargeAddressAware>
     <TargetFramework>netcoreapp2.0</TargetFramework>
-    <RuntimeIdentifiers>win7-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;osx.10.12-x64</RuntimeIdentifiers>
     <PackageTargetFallback>portable-net452;dotnet;netstandard1.6</PackageTargetFallback>
-    <SelfContained>true</SelfContained>
+    <RuntimeIdentifiers>win7-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;osx.10.12-x64</RuntimeIdentifiers>
+    <RuntimeIdentifier>$(RoslynRuntimeIdentifier)</RuntimeIdentifier>
+    <SelfContained Condition="'$(RoslynRuntimeIdentifier)' != ''" >true</SelfContained>
```
Comment (agocke) [inline src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj]: Not sure I understand the root cause here -- does SelfContained not work on older builds? Wouldn't always setting it also break things on dev machines?

Comment (jaredpar) [inline src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj]: It used to work incorrectly.  It just so happens that the version of VS that installed on Roslyn team members boxes today corrected the bug and now errors.  And yes it happened on the day we merged the SDK into master.  :rage1:

**Comment by jaredpar** [comment]
retest windows_debug_unit32_prtest please

**Pull request merged**